### ISSUE TITLE
Taxes for fees are calculated twice when using $order->calculate_taxes();

### ISSUE THREAD

**Issue by FlorianRachor** [woothemes/woocommerce#11103]
## EXPLANATION OF THE ISSUE

I created a new status for overdue payments. Hooking into "woocommerce_order_status_changed" I add a fee when the order changes into this status using $order->add_fee( $fee_properties );

I set the fee to taxable (full code below) and afterwards run $order->calculate_totals(); to update the amounts.

The problem is: The tax total will be calculated wrong, including the taxes for the fee two times. Note: Only the tax total, every other total will be fine. Here is an example:

(net price / tax, tax rate is 20%)
Produkt one: 20,83 € / 4,17 €
Product two: 20,83 € / 4,17 €
Shipping: 10 € / 2 €
Mahnspesen (fee): 33,33 € / 6,66 €

Total Shipping: 10€
Taxes: 23,67 (This includes TWICE the taxes for fee)
Total: 102 (Oddly enough, the total sum is correct)

HOWEVER: When I click "Update taxes & total" in the Wordpress Backend, the tax total will be calculated correct. So this has something to do with WC_Order::calculates_taxes();

I tried this with different versions of WooCommerce on different servers with the same result, so I guess it should be issue.

## STEPS TO REPRODUCE THE ISSUE
Full Code I used:

```
add_action( "woocommerce_order_status_changed", "user_inkasso_action" );
function user_inkasso_action( $order_id, $checkout = null ) {
    global $woocommerce;
    $order = wc_get_order( $order_id );

        $name      = 'Random Fee';
        $amount    = '33.33';
        $tax_class = '';
        $taxable   = true;

        $new_fee            = new stdClass();
        $new_fee->id        = sanitize_title( $name );
        $new_fee->name      = esc_attr( $name );
        $new_fee->amount    = (float) esc_attr( $amount );
        $new_fee->tax_class = $tax_class;
        $new_fee->taxable   = $taxable ? true : false;
        $new_fee->tax       = 0;
        $new_fee->tax_data  = array();

        $order->add_fee( $new_fee );
        $order->calculate_totals();
}
```

**Comment by mikejolley** [comment]
Do you think you can wrap your code correctly above, or post to gist.github.com? Also, using WC_Order class directly is not recommended. Use wc_get_order(). Also, what is 'inkasso' status? Thats not core.

**Comment by FlorianRachor** [comment]
I wrapped the code, and as I wrote I added a new status. I can change the the way I add the fee, but I highly doubt this has something to do with that, since the calculation happens after the fee is added.

If I run "calculate_totals()" on every order in the store separately I get the same result (no matter if the order is still in my custom status or any other status.  But for good measure I'll try with wc_get_order() on our dev server and see if it makes a difference.

**Comment by mikejolley** [comment]
Fees are calculating with your code on my store fine, however, shipping taxes are not being added for some reason. Debugging now.

**Comment by mikejolley** [comment]
Testing in 2.6 after fixing the shipping bug,

before status change: https://dl.dropboxusercontent.com/s/18o1zvst7t8ohce/2016-06-13%20at%2017.45.png?dl=0

after status change: https://dl.dropboxusercontent.com/s/0mt85z8j6a0yh69/2016-06-13%20at%2017.45%20%281%29.png?dl=0

So seems to be ok.

**Comment by mikejolley** [comment]
Repeated test in 2.5.5 and again, totals were fine after changing status https://dl.dropboxusercontent.com/s/irfjaq9vyfvzlvm/2016-06-13%20at%2017.47.png?dl=0

Perhaps you're seeing a conflict with another plugin such as WPML - I don't know what you're running on your site.

**Comment by FlorianRachor** [comment]
O.K thank you, I'll see if I can rule out a conflict with another plugin.

**Comment by krzysztofp** [comment]
@FlorianRachor  - do you remember what was the issue? I'm having the same double tax error today. Using WPML and custom theme with around 30 other plugins.

**Comment by willgorham** [comment]
@krzysztofp What you're seeing is probably a new unrelated issue: #17147 

The fix is in #17149
